collections/soldiers.js

```diff
@@ -28,4 +27,28 @@ function handleInsertion(soldierData, done) {
   }
 }
 
+function handleFind(soldierId, soldierName, done) {
+  MongoClient.connect(url, function (err, client) {
```

Comment:
Instead of creating a new connection and closing it for each query, you can create 1 connection and keep it open throughout your program. This both cleans out your code & improves performance.